analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/RestrictDeserializedTypesBase.cs

```diff
@@ -80,7 +81,7 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
             var additionalLocations = methodDeclaration is not null
                 ? new[] { GetIdentifier(methodDeclaration).GetLocation() }
                 : Array.Empty<Location>();
-            ReportIssue(operation, additionalLocations, RestrictTypesMessage);
+            ReportIssue(operation, additionalLocations, additionalLocations.ToProperties(SecondaryMessage), RestrictTypesMessage);
```

Comment:
Educational: I don't understand how the link between the secondary locations and the messages in the properties is done. The MS documentation is not clear about this. How is it done?